File: C++/2434.cpp
#include <iostream>

using namespace std;

int main(){
    int n, init, dh, lowest_value, i;
    cin >> n >> init;
    lowest_value = init;
    
    for(i = 0; i < n; i++){
        cin >> dh;
        init += dh;
        if(init < lowest_value) lowest_value = init;
    }
    
    cout << lowest_value << endl;
    return 0;
}

File: javascript/2434.js
var input = require('fs').readFileSync('/dev/stdin', 'utf8');
var lines = input.split('\n');
[n, init] = lines.shift().split(' ').map(e => parseInt(e))
var lowest_value = init

for(let i = 0; i < n; i++) {
    var dh = parseInt(lines.shift())
    init += dh

    if(init < lowest_value) lowest_value = init
}

console.log(lowest_value)